waku/v2/node/connectedness.go

```diff
@@ -160,20 +161,38 @@ func (w *WakuNode) Status() (isOnline bool, hasHistory bool) {
 			if !hasStore && protocol == store.StoreID_v20beta4 {
 				hasStore = true
 			}
-			if !hasFilter && protocol == legacy_filter.FilterID_v20beta1 {
+			if !hasFilter && protocol == filter.FilterSubscribeID_v20beta1 {
 				hasFilter = true
 			}
 		}
 	}
 
 	if hasStore {
 		hasHistory = true
+	} else {
+		//Look if there are other peers in peerStore that supports Store and we are not connected to them
+		for _, peer := range w.host.Peerstore().Peers() {
+			protocols, err := w.host.Peerstore().GetProtocols(peer)
+			if err != nil {
+				w.log.Warn("reading peer protocols", logging.HostID("peer", peer), zap.Error(err))
+				continue
+			}
+			for _, protocol := range protocols {
+				if !hasStore && protocol == store.StoreID_v20beta4 {
+					hasStore = true
+					break
+				}
+			}
+			if hasStore {
+				break
+			}
+		}
 	}
 
 	if w.opts.enableFilterLightNode && !w.opts.enableRelay {
 		isOnline = hasLightPush && hasFilter
 	} else {
-		isOnline = hasRelay || hasLightPush && (hasStore || hasFilter)
+		isOnline = hasRelay
```

Comment:
This change makes sense to do now. The reason why it checked for store before was because we always kept a connection to a store node (this code was added particularly for status-go back then). Not sure why we checked for filter tho? Maybe @vitvly remembers!